src/bootstrap/doc.rs

```diff
@@ -582,18 +582,30 @@ impl Step for Rustc {
 
         // Build cargo command.
         let mut cargo = builder.cargo(compiler, Mode::Rustc, SourceType::InTree, target, "doc");
+
+        cargo.rustdocflag("-Zunstable-options");
         cargo.rustdocflag("--document-private-items");
         // Since we always pass --document-private-items, there's no need to warn about linking to private items.
         cargo.rustdocflag("-Arustdoc::private-intra-doc-links");
         cargo.rustdocflag("--enable-index-page");
-        cargo.rustdocflag("-Zunstable-options");
         cargo.rustdocflag("-Znormalize-docs");
         cargo.rustdocflag("--show-type-layout");
+
         compile::rustc_cargo(builder, &mut cargo, target);
+
+        cargo.arg("-Zunstable-options");
```

Comment:
nit: these changes look unnecessary for the most part

Reply:
they make it more readable for me, though that's just personal preference

can revert if you want

Reply:
Yeah, I would rather revert if you don't mind.